tutorials/scripting/groups.rst

```diff
@@ -0,0 +1,124 @@
+.. _doc_groups:
+
+Groups
+======
+
+Groups in Godot work like tags in other software. You can add a node to as many
+groups as you want. Then, in code, you can use the SceneTree to:
+
+- Get a list of nodes in a group.
+- Call a method on all nodes in a group.
+- Send a notification to all nodes in a group.
+
+This is a useful feature to organize large scenes and decouple code.
+
+Adding nodes to a group
+-----------------------
+
+There are two ways to add nodes to groups:
+
+- Using the Node dock in the editor.
+- Calling :ref:`SceneTree.call_group() <class_SceneTree_method_call_group>`.
+
+Using the Node dock
+~~~~~~~~~~~~~~~~~~~
+
+You can add nodes in the current scene to groups using the Groups tab in the
+Node dock.
+
+.. image:: img/groups_node_tab.png
+
+Select one or more nodes in the Scene dock and write the group name in the
+field, then click Add.
+
+.. image:: img/groups_add_node_to_group.png
+
+You should now see the group appear.
+
+.. image:: img/groups_node_after_adding.png
+
+In a complex project, you may end up with many groups or large scenes with many
+nodes. You can add or remove any node to groups using the Group Editor window.
+To access it, click the Manage Groups button.
+
+.. image:: img/groups_manage_groups_button.png
+
+The Group Editor window appears. Here's a screenshot from a complex project to
+illustrate the tool's purpose.
+
+.. image:: img/groups_group_editor_window.png
+
+It has three columns:
+
+1. A list of groups used by nodes in the current scene.
+2. A list of nodes that are not part of the selected group.
+3. A list of nodes in the group.
+
+The fields at the bottom allow you to add new groups or filter nodes in the
+second and third columns.
+
+.. note:: Any node name that's greyed out means the node was added to the group
+          in a different scene and you cannot edit it here. This happens on
+          scene instances in particular.
+
+Using code
+~~~~~~~~~~
+
+You can also manage groups from scripts. The following code adds the node to
+which you attach the script to the ``guards`` group as soon as it entered the
```

Comment:
Ready happens after the node entered the scene tree. It's as soon as it finished entering or something like that. Is it grammatically incorrect with the past tense like this?

Reply:
It would be "enters" or "has entered". Your sentence is not past tense with words like "adds" rather than "added" - you're talking about what it is doing rather than what it did.